root/user/ContactUser.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * @flow strict
+ * Copyright (C) 2022 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import UserAccountLayout, {
+  type AccountLayoutUserT,
+} from '../components/UserAccountLayout.js';
+import FormCsrfToken
+  from '../static/scripts/edit/components/FormCsrfToken.js';
+import FormRow from '../static/scripts/edit/components/FormRow.js';
+import FormRowCheckbox
+  from '../static/scripts/edit/components/FormRowCheckbox.js';
+import FormRowTextArea
+  from '../static/scripts/edit/components/FormRowTextArea.js';
+import FormRowTextLong
+  from '../static/scripts/edit/components/FormRowTextLong.js';
+import FormSubmit from '../static/scripts/edit/components/FormSubmit.js';
+
+
+type Props = {
+  +form: FormT<{
+    +body: FieldT<string>,
+    +csrf_token: FieldT<string>,
+    +reveal_address: FieldT<boolean>,
+    +send_to_self: FieldT<boolean>,
+    +subject: FieldT<string>,
+  }>,
+  +user: AccountLayoutUserT,
+};
+
+const ContactUser = ({
+  form,
+  user,
+}: Props): React.Element<typeof UserAccountLayout> => (
+  <UserAccountLayout
+    entity={user}
+    page="report"
+    title={l('Send Email')}
+  >
+    <h2>{l('Send Email')}</h2>
+
+    <form className="contact-form" method="post">
+      <FormCsrfToken form={form} />
```

Comment:
Apparently `Form::User::Contact` is missing the `Form::Role::CSRFToken` role, so this doesn't actually do anything. But that was a problem with the TT page too.

Reply:
@mwiencek: Shouldn’t this role be added to this class then? As a separate ticket/pull request?

Reply:
I added it in a second commit. @mwiencek: Could you test it in your local setup again to make sure it works as expected? :)

Reply:
It reloads the form now if the token is invalid, but doesn't display any error message. Probably because you are calling `$form->TO_JSON` before the form is validated.

Reply:
I understand this just means we need to move the Node stash data setting below the `if ($c->form_posted_and_valid($form))` block, so I did that now.